internal/services/discovery/openshift_test.go

```diff
@@ -0,0 +1,90 @@
+package discovery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+
+	"castai-agent/internal/services/controller/scheme"
+)
+
+func TestServiceImpl_GetOpenshiftClusterID(t *testing.T) {
+	r := require.New(t)
+
+	clusterVersionJSON := `
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+  uid: dc0570f9-2c46-40b1-a5d0-c7233e82c7b6
+spec:
+  clusterID: 91d87440-5173-47f0-aca5-65bc0144ad30
+`
+
+	var clusterVersionMap map[string]any
+	r.NoError(yaml.Unmarshal([]byte(clusterVersionJSON), &clusterVersionMap))
+
+	version := &unstructured.Unstructured{}
+	version.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   OpenshiftClusterVersionsGVR.Group,
+		Version: OpenshiftClusterVersionsGVR.Version,
+		Kind:    "ClusterVersion",
+	})
+	version.Object = clusterVersionMap
+
+	clientset := fakeclientset.NewSimpleClientset(version)
+	dyno := fakedynamic.NewSimpleDynamicClient(scheme.Scheme, version)
+
+	s := New(clientset, dyno)
+
+	internalID, err := s.GetOpenshiftClusterID(context.Background())
+
+	r.NoError(err)
+	r.Equal("91d87440-5173-47f0-aca5-65bc0144ad30", *internalID)
+}
+
+func TestServiceImpl_GetOpenshiftClusterName(t *testing.T) {
+	r := require.New(t)
+
+	masterMachineJSON := `
+apiVersion: machine.openshift.io/v1beta1
+kind: Machine
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: foo-bar
+    machine.openshift.io/cluster-api-machine-role: master
+    machine.openshift.io/cluster-api-machine-type: master
+    machine.openshift.io/instance-type: m5.2xlarge
+    machine.openshift.io/region: eu-central-1
+    machine.openshift.io/zone: eu-central-1a
+  name: foo-bar-master-0
+  namespace: openshift-machine-api
+  uid: 803fbff1-bab4-412c-912b-0ba7f0ef1dcd
+`
+	var masterMachineMap map[string]any
+	r.NoError(yaml.Unmarshal([]byte(masterMachineJSON), &masterMachineMap))
+
+	masterMachine := &unstructured.Unstructured{}
+	masterMachine.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   OpenshiftMachinesGVR.Group,
+		Version: OpenshiftMachinesGVR.Version,
+		Kind:    "Machine",
+	})
+	masterMachine.Object = masterMachineMap
```

Comment:
is this this string-toYAML-to-generic-structure needed? can we just construct an object here?:

```suggestion
	masterMachine.Object = GenericJson{
	   "apiVersion": "machine.openshift...."
	   ...
	   "metadata": GenericJson{
	     "labels": GenericJson {
	         "machine.openshift.io/cluster-api-cluster": "foo-bar"
	     }
	   }
	}
```

Reply:
maybe it's not pretty either, when we don't have the golang struct for the "object"